lib/migration.js

```diff
@@ -41,14 +41,14 @@ class Migration {
   }
 
   _apply(action, pgm) {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       if (action.length === 2) {
         action(pgm, resolve);
       } else {
         const result = action(pgm);
         // result conforms to Promises/A+ spec
         if (typeof result === 'object' && typeof result.then === 'function') {
-          result.then(resolve);
+          result.then(resolve).catch(err => reject(err));
```

Comment:
Thanks for contributing!
I would prefer `.catch(reject)`

Reply:
It's not the same thing!

In fact if you `.catch(reject)` the error isn't passed to the reject call and thus you won't get the error message for the rejection.

Instead if you do `.catch(err => reject(err))` you pass the error and when the rejection is handled it will print out the error message to the user.

Reply:
It is almost the same thing as `.catch(reject)` is equivalent to `.catch((...args) => reject(...args))`

Reply:
Try the difference in your code

Reply:
Tried and got same result

Reply:
```
Promise.reject(new Error('fail')).catch(console.log.bind(console));
Promise.reject(new Error('fail')).catch(err => console.log(err));
```
have same result...

Reply:
Ok